src/Features/Core/Portable/TodoComments/ITodoCommentService.cs

```diff
@@ -38,37 +37,34 @@ public TodoComment(TodoCommentDescriptor descriptor, string message, int positio
             var location = tree == null
                 ? Location.Create(document.FilePath!, textSpan, text.Lines.GetLinePositionSpan(textSpan))
                 : tree.GetLocation(textSpan);
-            var originalLineInfo = location.GetLineSpan();
-            var mappedLineInfo = location.GetMappedLineSpan();
 
-            return new(
-                priority: Descriptor.Priority,
-                message: Message,
-                documentId: document.Id,
-                originalLine: originalLineInfo.StartLinePosition.Line,
-                originalColumn: originalLineInfo.StartLinePosition.Character,
-                originalFilePath: document.FilePath,
-                mappedLine: mappedLineInfo.StartLinePosition.Line,
-                mappedColumn: mappedLineInfo.StartLinePosition.Character,
-                mappedFilePath: mappedLineInfo.GetMappedFilePathIfExist());
+            return new TodoCommentData(
+                Descriptor.Priority,
+                Message,
+                document.Id,
+                location.GetLineSpan(),
+                location.GetMappedLineSpan());
```

Comment:
moving from 6 fields to 2 FileLinePositionSpan instances that represent the same.